src/main/java/com/moodybluez/enterprise/dto/Date.java

```diff
@@ -5,6 +5,7 @@
 public @Data
 class Date {
```

Comment:
To be honest, I was a bit confused as to why have a separate class for Date, but after carefully looking at your project goals, I see how it can play important role when you try to get metrics.

Reply:
1) Don't name the class Date, as that name is already taken... twice:
java.util.Date
java.sql.Date

One more date is confusing.

2) Can you get all of this information from Calendar/GregorianCalendar?  More than likely, yes.  If so, creating a new class to do the same thing is redundant.